ParquetSharp.DataFrame/DataFrameExtensions.cs

```diff
@@ -8,58 +8,100 @@ namespace ParquetSharp
         /// </summary>
-        /// <param name="dataFrame">The DataFrame to write</param>
+        /// <param name="dataFrame">DataFrame to write.</param>
+        /// <param name="fileWriter">Writer to use.</param>
+        /// <param name="rowGroupSize">Maximum number of rows per row group</param>
+        public static void ToParquet(this DataFrame dataFrame, ParquetFileWriter fileWriter, int rowGroupSize = 1024 * 1024)
+        {
+            long numRows = dataFrame.Rows.Count;
+            long offset = 0L;
+
+            while (offset < numRows)
+            {
+                int batchSize = (int)Math.Min(numRows - offset, rowGroupSize);
+                using RowGroupWriter rowGroupWriter = fileWriter.AppendRowGroup();
+                foreach (DataFrameColumn dataFrameColumn in dataFrame.Columns)
+                {
+                    using ColumnWriter columnWriter = rowGroupWriter.NextColumn();
+                    using LogicalColumnWriter logicalWriter = columnWriter.LogicalWriter();
+                    logicalWriter.Apply(new DataFrameWriter(dataFrameColumn, offset, batchSize));
+                }
+
+                offset += batchSize;
+            }
+        }
+
+        /// <summary>
+        /// Write DataFrames in Parquet format
+        /// </summary>
+        /// <param name="dataFrame">DataFrames to write</param>
         /// <param name="path">Path to write to</param>
         /// <param name="writerProperties">Optional writer properties that override the default properties</param>
         /// <param name="logicalTypeOverrides">Mapping from column names to Parquet logical types,
         /// overriding the default logical types. When writing decimal columns, a logical type must be provided
         /// to specify the precision and scale to use.</param>
         /// <param name="rowGroupSize">Maximum number of rows per row group</param>
         public static void ToParquet(
-            this DataFrame dataFrame, string path, WriterProperties? writerProperties = null,
+            this IEnumerable<DataFrame> dataFrames, string path, WriterProperties? writerProperties = null,
             IReadOnlyDictionary<string, LogicalType>? logicalTypeOverrides = null, int rowGroupSize = 1024 * 1024)
         {
-            var schemaColumns = dataFrame.Columns.Select(col => GetSchemaColumn(
-                col, logicalTypeOverrides != null && logicalTypeOverrides.TryGetValue(col.Name, out var logicalType) ? logicalType : null)).ToArray();
-            using var fileWriter = writerProperties == null
-                ? new ParquetFileWriter(path, schemaColumns)
-                : new ParquetFileWriter(path, schemaColumns, writerProperties);
+            if (dataFrames is null)
+                throw new ArgumentNullException(nameof(dataFrames));
 
-            var numRows = dataFrame.Rows.Count;
-            var offset = 0L;
+            DataFrame firstDataFrame = dataFrames.First();
```

Comment:
Using `First()` then a `foreach` over dataFrames below requires multiple enumerations over the `IEnumerable`, which won't be compatible with all `IEnumerable` types. I think you'll probably want to create a null `ParquetFileWriter` and then assign it on the first iteration of the loop.